### PR TITLE
Update to 3.80

### PR DESCRIPTION
**Pull request by mwittgen**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

closes #12

### CODE REVIEW

**Comment by mwittgen** [comment]
This PR simplifies the build scripts from https://github.com/conda-forge/healpix_cxx-feedstock/pull/12/
The most hacks in the original 3.31 feedstock have been fixed in  https://github.com/conda-forge/healpix_cxx-feedstock/pull/15

**Comment by mwittgen** [comment]
This PR is meant to be merged after #15 and rebase.

**Review by beckermr — CHANGES_REQUESTED** [review]
This configuration will likely build and ship a libsharp.so or dylib. We were trying to avoid that in the previous PR which is why it got complicated.

**Comment by mwittgen** [comment]
> This configuration will likely build and ship a libsharp.so or dylib. We were trying to avoid that in the previous PR which is why it got complicated.

less   build_artifacts/linux-64/healpix_cxx-3.80-h08efbc1_0.tar.bz2  | cat
-rw-r--r-- 1000/1000        42 2022-01-05 17:26 info/run_exports.json
-rw-r--r-- 1000/1000       206 2022-01-05 17:26 info/hash_input.json
-rw-r--r-- 1000/1000       357 2022-01-05 17:26 info/index.json
-rw-r--r-- 1000/1000       577 2022-01-05 17:26 info/has_prefix
-rw-r--r-- 1000/1000      1702 2022-01-05 17:26 info/files
-rw-r--r-- 1000/1000      5826 2022-01-05 17:26 info/about.json
-rw-r--r-- 1000/1000     12838 2022-01-05 17:26 info/paths.json
-rw-r--r-- 1000/1000         0 2022-01-05 17:26 info/git
-rw-rw-r-- 1000/1000      3437 2022-01-05 17:23 info/recipe/meta.yaml.template
-rw-rw-r-- 1000/1000       664 2022-01-05 17:22 info/recipe/build.sh
-rw-r--r-- 1000/1000      1616 2022-01-05 17:26 info/test/run_test.sh
-rw-r--r-- 1000/1000       777 2022-01-05 17:26 info/recipe/conda_build_config.yaml
-rw-r--r-- 1000/1000      4916 2022-01-05 17:26 info/recipe/meta.yaml
-rw-rw-r-- 1000/1000       336 2022-01-05 17:26 include/healpix_cxx/alice3.h
-rw-rw-r-- 1000/1000      1304 2022-01-05 17:26 include/healpix_cxx/mask_tools.h
-rw-rw-r-- 1000/1000      1403 2022-01-05 17:26 include/healpix_cxx/xcomplex.h
-rw-rw-r-- 1000/1000      1582 2022-01-05 17:26 include/healpix_cxx/moc_fitsio.h
-rw-rw-r-- 1000/1000      1957 2022-01-05 17:26 include/healpix_cxx/healpix_tables.h
-rw-rw-r-- 1000/1000      2014 2022-01-05 17:26 include/libsharp/sharp_almhelpers.h
-rw-rw-r-- 1000/1000      2101 2022-01-05 17:26 include/healpix_cxx/healpix_data_io.h
-rw-rw-r-- 1000/1000      2639 2022-01-05 17:26 include/healpix_cxx/pointing.h
-rw-rw-r-- 1000/1000      2669 2022-01-05 17:26 include/healpix_cxx/alloc_utils.h
-rw-rw-r-- 1000/1000      2757 2022-01-05 17:26 include/healpix_cxx/safe_cast.h
-rw-rw-r-- 1000/1000      2777 2022-01-05 17:26 include/healpix_cxx/powspec_fitsio.h
-rw-rw-r-- 1000/1000      2912 2022-01-05 17:26 include/healpix_cxx/geom_utils.h
-rw-rw-r-- 1000/1000      2918 2022-01-05 17:26 include/healpix_cxx/moc_query.h
-rw-rw-r-- 1000/1000      2982 2022-01-05 17:26 include/healpix_cxx/lsconstants.h
-rw-rw-r-- 1000/1000      3014 2022-01-05 17:26 include/healpix_cxx/weight_utils.h
-rw-rw-r-- 1000/1000      3119 2022-01-05 17:26 include/healpix_cxx/error_handling.h
-rw-rw-r-- 1000/1000      4197 2022-01-05 17:26 include/healpix_cxx/planck_rng.h
-rw-rw-r-- 1000/1000      4547 2022-01-05 17:26 include/healpix_cxx/healpix_map_fitsio.h
-rw-rw-r-- 1000/1000      4571 2022-01-05 17:26 include/healpix_cxx/powspec.h
-rw-rw-r-- 1000/1000      4606 2022-01-05 17:26 include/healpix_cxx/string_utils.h
-rw-rw-r-- 1000/1000      4796 2022-01-05 17:26 include/healpix_cxx/rotmatrix.h
-rw-rw-r-- 1000/1000      4817 2022-01-05 17:26 include/healpix_cxx/vec3.h
-rw-rw-r-- 1000/1000      5237 2022-01-05 17:26 include/healpix_cxx/alm_powspec_tools.h
-rw-rw-r-- 1000/1000      5733 2022-01-05 17:26 include/healpix_cxx/alm_fitsio.h
-rw-rw-r-- 1000/1000      5982 2022-01-05 17:26 include/healpix_cxx/wigner.h
-rw-rw-r-- 1000/1000      6100 2022-01-05 17:26 include/healpix_cxx/compress_utils.h
-rw-rw-r-- 1000/1000      6122 2022-01-05 17:26 include/healpix_cxx/alm.h
-rw-rw-r-- 1000/1000      7085 2022-01-05 17:26 include/libsharp/sharp_geomhelpers.h
-rw-rw-r-- 1000/1000      7192 2022-01-05 17:26 include/healpix_cxx/moc.h
-rw-rw-r-- 1000/1000      7848 2022-01-05 17:26 include/libsharp/sharp_cxx.h
-rw-rw-r-- 1000/1000      8542 2022-01-05 17:26 include/healpix_cxx/math_utils.h
-rw-rw-r-- 1000/1000      8646 2022-01-05 17:26 include/healpix_cxx/alm_healpix_tools.h
-rw-rw-r-- 1000/1000     10486 2022-01-05 17:26 include/healpix_cxx/healpix_map.h
-rw-rw-r-- 1000/1000     10825 2022-01-05 17:26 include/libsharp/sharp.h
-rw-rw-r-- 1000/1000     11169 2022-01-05 17:26 include/healpix_cxx/datatypes.h
-rw-rw-r-- 1000/1000     12529 2022-01-05 17:26 include/healpix_cxx/fitshandle.h
-rw-rw-r-- 1000/1000     12947 2022-01-05 17:26 include/healpix_cxx/sse_utils_cxx.h
-rw-rw-r-- 1000/1000     14548 2022-01-05 17:26 include/healpix_cxx/rangeset.h
-rw-rw-r-- 1000/1000     20580 2022-01-05 17:26 include/healpix_cxx/healpix_base.h
-rw-rw-r-- 1000/1000     23939 2022-01-05 17:26 include/healpix_cxx/arr.h
lrwxrwxrwx 1000/1000         0 2022-01-05 17:26 lib/libhealpix_cxx.so -> libhealpix_cxx.so.2.0.1
lrwxrwxrwx 1000/1000         0 2022-01-05 17:26 lib/libhealpix_cxx.so.2 -> libhealpix_cxx.so.2.0.1
-rw-rw-r-- 1000/1000       663 2022-01-05 17:26 lib/pkgconfig/libsharp.pc
-rw-rw-r-- 1000/1000      1225 2022-01-05 17:26 lib/pkgconfig/healpix_cxx.pc
-rw-r--r-- 1000/1000     18011 2010-11-24 07:14 info/licenses/COPYING
-rw-r--r-- 1000/1000     18092 2022-01-05 17:26 info/licenses/COPYING-libsharp
-rwxrwxr-x 1000/1000     23984 2022-01-05 17:26 bin/rotalm_cxx
-rwxrwxr-x 1000/1000     27304 2022-01-05 17:26 bin/calc_powspec
-rwxrwxr-x 1000/1000     27464 2022-01-05 17:26 bin/median_filter_cxx
-rwxrwxr-x 1000/1000     27936 2022-01-05 17:26 bin/syn_alm_cxx
-rwxrwxr-x 1000/1000     28784 2022-01-05 17:26 bin/hotspots_cxx
-rwxrwxr-x 1000/1000     37152 2022-01-05 17:26 bin/alice3
-rwxrwxr-x 1000/1000     37496 2022-01-05 17:26 bin/compute_weights
-rwxrwxr-x 1000/1000     43024 2022-01-05 17:26 bin/smoothing_cxx
-rwxrwxr-x 1000/1000     46104 2022-01-05 17:26 bin/udgrade_cxx
-rwxrwxr-x 1000/1000     46848 2022-01-05 17:26 bin/anafast_cxx
-rwxrwxr-x 1000/1000     49488 2022-01-05 17:26 bin/mult_alm
-rwxrwxr-x 1000/1000     49864 2022-01-05 17:26 bin/alm2map_cxx
-rwxrwxr-x 1000/1000     54968 2022-01-05 17:26 bin/udgrade_harmonic_cxx
-rwxrwxr-x 1000/1000     57952 2022-01-05 17:26 bin/map2tga
-rwxrwxr-x 1000/1000     64752 2022-01-05 17:26 bin/needlet_tool
-rw-rw-r-- 1000/1000     77760 2022-01-05 02:14 info/recipe/cl.fits
-rw-rw-r-- 1000/1000     77760 2022-01-05 02:14 info/test/cl.fits
-rwxrwxr-x 1000/1000    950648 2022-01-05 17:26 lib/libhealpix_cxx.so.2.0.1

**Comment by beckermr** [comment]
Ideally we'd build libsharp in the libsharp-feedstock and then link this build against the libsharp.so built there. However, I was told there was some issue with this.

**Comment by mwittgen** [comment]
I'll check the Mac version as well. libsharp is not appearing in the package. What is evidence for your claims?

**Comment by beckermr** [comment]
It was in previous builds AFAIK.

**Comment by beckermr** [comment]
I do a librsharp header:

> -rw-rw-r-- 1000/1000 7848 2022-01-05 17:26 include/libsharp/sharp_cxx.h

**Comment by mwittgen** [comment]
> Ideally we'd build libsharp in the libsharp-feedstock and then link this build against the libsharp.so built there. However, I was told there was some issue with this.
I see that's broken in libsharp from the feedstock

mamba install libsharp
find $CONDA_PREFIX -name "*sharp*"
/home/wittgen/lsst/lsst_stack/conda/envs/lsst-scipipe-0.7.0/include/sharp_legendre.h
/home/wittgen/lsst/lsst_stack/conda/envs/lsst-scipipe-0.7.0/include/sharp_announce.h
/home/wittgen/lsst/lsst_stack/conda/envs/lsst-scipipe-0.7.0/include/sharp_lowlevel.h
/home/wittgen/lsst/lsst_stack/conda/envs/lsst-scipipe-0.7.0/include/sharp_geomhelpers.h
/home/wittgen/lsst/lsst_stack/conda/envs/lsst-scipipe-0.7.0/include/sharp_core.h
...
/home/wittgen/lsst/lsst_stack/conda/envs/lsst-scipipe-0.7.0/lib/python3.8/site-packages/libsharp/libsharp.cpython-38-x86_64-linux-gnu.so
/home/wittgen/lsst/lsst_stack/conda/envs/lsst-scipipe-0.7.0/lib/libsharp.a

ldd /home/wittgen/lsst/lsst_stack/conda/envs/lsst-scipipe-0.7.0/lib/python3.8/site-packages/libsharp/libsharp.cpython-38-x86_64-linux-gnu.so
/home/wittgen/lsst/lsst_stack/conda/envs/lsst-scipipe-0.7.0/lib/python3.8/site-packages/libsharp/libsharp.cpython-38-x86_64-linux-gnu.so

So the python lib has libsharp statically linked. Ideally this would be shared lib and healpix_cxx could depend on libsharp

**Comment by beckermr** [comment]
This is looking good. all we need is a run export and we are done. See the other PR.

File: recipe/build.sh [hunk 1]
```diff
@@ -1,34 +1,28 @@
-cd src/cxx
-
-autoconf
+# Build libsharp
+cd src/common_libraries/libsharp
 
-export CPATH="${PREFIX}/include"
+# Add -fPIC flag so libsharp.a can be linked into libhealpix_cxx later
+CXXFLAGS=-fPIC CFLAGS=-fPIC ./configure --prefix=$PREFIX --disable-silent-rules --disable-dependency-tracking --disable-shared
```
Comment (beckermr) [inline recipe/build.sh:5]: We need to append these flags to the CXXFLAGS and CFLAGS provided by conda.

**Review by beckermr — COMMENTED** [review]
One more thing.

**Comment by mwittgen** [comment]
I'll test it with a use case later.

**Comment by beckermr** [comment]
This will need to be rebased and then we can merge in a few hours.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Review by beckermr — APPROVED** [review]

**Pull request merged**